pkg/registry/event/rest.go

```diff
@@ -45,7 +46,8 @@ func NewStorage(registry generic.Registry) *REST {
 func (rs *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, error) {
 	event, ok := obj.(*api.Event)
 	if !ok {
-		return nil, fmt.Errorf("invalid object type")
+		glog.V(2).Infof("Invalid object type. Was expecting an object of type api.Event: %v", obj)
```

Comment:
Surfacing internal details in user errors to enable debugging doesnt seem right to me.
We should return a simple error that indicates something is wrong. We should then be able to figure out the problem, with repro steps, by looking at the logs.

This is a programmer error. User errors are caught here: https://github.com/GoogleCloudPlatform/kubernetes/blob/141eba39e93ef3f8fe53ce09405a5b83db9737a4/pkg/apiserver/resthandler.go#L305